### PR TITLE
OIA-37: Update OIA eventconf to support event-sourced data collection

### PR DESCRIPTION
**Pull request by freddy61025**
https://issues.opennms.org/browse/OIA-37

Tested 0.8.4 with opennms then roll back to 1.0.0
https://app.circleci.com/pipelines/github/OpenNMS/opennms?branch=jira%2FOIA-37-smoke&filter=all

### CODE REVIEW

File: config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2010-2016 The OpenNMS Group, Inc.
```
Comment (cgorantla) [inline config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java:4]:
```suggestion
 * Copyright (C) 2022 The OpenNMS Group, Inc.
```

File: config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * OpenNMS(R) is Copyright (C) 1999-2016 The OpenNMS Group, Inc.
```
Comment (cgorantla) [inline config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java:5]:
```suggestion
 * OpenNMS(R) is Copyright (C) 2022 The OpenNMS Group, Inc.
```

File: config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+package org.opennms.integration.api.xml.schema.eventconf;
+
+import org.opennms.integration.api.xml.schema.eventconf.adpaters.AttributeTypeAdapter;
+
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+/**
+ * Defines how a particular attribute should be persisted.
+ *
+ * @author jwhite
```
Comment (cgorantla) [inline config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java:38]: Leave out author or add yourself.

**Review by cgorantla — COMMENTED** [review]
Minor comments.

File: config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java (hunk 2 above)
Comment (freddy61025) [reply config/src/main/java/org/opennms/integration/api/xml/schema/eventconf/AttributeType.java]: should it be OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.?

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**